usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -7545,6 +8570,30 @@ static int read_adapter_config_file(STDLL_TokData_t * tokdata,
                 ep11_data->digest_libica_path[sizeof(ep11_data->digest_libica_path)-1] = '\0';
             }
             i = 0;
+        } else if (i == 6) {
+            /* expecting pkey_mode */
+            if (token == NULL) {
+                rc = APQN_FILE_UNEXPECTED_END_OF_FILE;
+                OCK_SYSLOG(LOG_ERR,"%s: Error: Unexpected end of file found"
+                           " in config file '%s', expected pkey_mode 0 .. 3\n",
+                           __func__, fname);
+                break;
+            }
+            if (strncmp(token, "DISABLED", 8) == 0)
```

Comment:
You never explicitely initialize ep11_data->pkey_mode, so it is 0 by default, because ep11_data is allocated using calloc, and thus is cleared with zeros. 
So this means it defaults to PKEY_MODE_DISABLED. Shouldn't it default to PKEY_MODE_DEFAULT instead? 
If so, you should probably initialize it to PKEY_MODE_DEFAULT  somewhere in ep11tok_init.

Reply:
Done.